crates/hir_def/src/body.rs

```diff
@@ -99,7 +93,7 @@ impl Expander {
         db: &dyn DefDatabase,
         macro_call: ast::MacroCall,
     ) -> Result<ExpandResult<Option<(Mark, T)>>, UnresolvedMacro> {
-        if self.recursion_limit + 1 > EXPANSION_RECURSION_LIMIT {
+        if self.recursion_limit + 1 > db.limits().expansion_recursion() {
```

Comment:
Note that we don’t want to write `>` manually here, and instead want to call some kind of `.check(value) -> Result<(),()>` function. That way, we’ll be able to quickly implement functionality like “print the maximal value that was still below the limit”.